trio/_path.py

```diff
@@ -446,6 +472,17 @@ async def with_segments(self, *pathsegments: StrPath) -> Path: ...
 
 """
 
+if sys.version_info < (3, 12):
+    # Since we synthesise methods from the stdlib, this automatically will
+    # have deprecation warnings, and disappear entirely in 3.12+.
+    Path.link_to.__doc__ = """
+        Like Python 3.8-3.11's :meth:`~pathlib.Path.link_to`, but async.
+
+    :deprecated: This method was deprecated in Python 3.10 and entirely \
+    removed in 3.12. Use :meth:`hardlink_to` instead which has \
+    a more meaningful parameter order.
+    """
+
```

Comment:
Why do we still overwrite doc with the new fix?

Reply:
Well it seems good to have the deprecation warning replicated in our docs, so users looking there will be redirected to the better function.